scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -629,15 +627,13 @@ def Mf32precond(x):
         listY = [Yf64, Yf32]
 
         tests = list(itertools.product(listA, listB, listM, listX, listY))
-        # This is one of the slower tests because there are >1,000 configs
-        # to test here, instead of checking product of all input, output types
-        # test each configuration for the first sparse format, and then
-        # for one additional sparse format. this takes 2/7=30% as long as
-        # testing all configurations for all sparse formats.
-        if s_f_i > 0:
-            tests = tests[s_f_i - 1::sparse_formats-1]
 
         for A, B, M, X, Y in tests:
+            # This is one of the slower tests because there are >1,000 configs
+            # to test here. Flip a biased coin to decide whether to run  each
+            # test to get decent coverage in less time.
+            if rnd.random() < 0.95:
+                continue  # too many tests
```

Comment:
This breaks a hard requirement that we have: tests should be reproducible. So this change cannot be right. Why not mark some subset as slow (or even xslow). Or, if achieving good test coverage in a reasonable amount of time really is that problematic, then this may be a fit for `hypothesis`. We default to fixed seeds, but it's possible with `hypothesis` to randomly sample the subspace in a more controlled way (plus if it fails, it tells you the reproducer).

Reply:
This has a fixed seed.